### ISSUE TITLE
Feature to set opt-in for 'background app run' instead of opt-out

### ISSUE THREAD

**Issue by kutysam** [d2phap/ImageGlass#1244]
Regarding `ImageGlass Kobe 8.4.1.22`
```
⚡ Breaking and Important changes
1. New app behaviors to set by default in Settings > General > Others:

Continue running background when the app is closed.
Start with OS.
```

I personally am very against this. Here are my reasons.

1. This is an opt-out feature and I don't think there is a notification for the user to notice it on launch. So for the retail users, they will not be aware of this
2. Yes, it definetly gives advantage to the app as it starts up fast but it actually slows down the com on startup of windows as it starts with OS.
3. I don't think many users use it daily or regularly and this definetly impacts them if they use a low memory computer as a part of memory is eaten up by imageglass.
4. If all apps adapt this approach, vscode, chrome, discord, telegram, whatsapp for example will startup with windows, the computer will definetly lag and users will have issues launching new apps on a low memory computer. --> This is the reason on why apps don't startup on windows except when it is really neccessary. 

What I would suggest is to 
1. Change it to opt-in.
2. On the launch of the next version, have a description of this feature and let the user select the option if they want to start it with the OS / always run in background.

Thank you.

**Comment by d2phap** [comment]
As announced, you can change these options in Settings > General or use config files to create your custom configuration if you want to opt out them when deploying to other machines.

https://imageglass.org/docs/app-configs

**Comment by kutysam** [comment]
Sorry, but I don't think you are replying to the root thing I am describing. As a retail (non-computer geek), they probably won't know there is such option to disable it as they don't meddle with settings.

TLDR
- By having it as opt out by default (without informing users that it is being enabled) and having it impacting the user's overall experience on the computer isn't a very right thing to do. (My startup impact is medium for your app in taskmgr)

- **I do see that you have 4 steps during installation. It will be good to include this under step 3, where there is a privacy portion and u can have checkboxes there to startup with windows etc.**

- If the idea is to get the app to open quicker as it loads up slow, I don't feel it is a valid argument as you are making users computers start up slow. If the app is something that someone uses everyday like telegram desktop, then it makes sense to start it up with the computer.

If you do feel otherwise, then feel free to keep it closed. All that I hope is that low-end cpu users get a better experience with their system startups.

**Comment by d2phap** [comment]
One of the advantages of open source software is the users can modify it to the way they want. In your case, I suggest you to modify the source code, change those settings in Source\Components\ImageGlass.Settings\Configs.cs, and redistribute it, and respect the GPLv3 license of ImageGlass, you need to disclose the source code, too.

For low-end CPU that cannot effort 30MB RAM of IG, can also stay in v8.3. I have been receiving many requests to modernize the app, but the reasons it's delayed for years because I want to let the users have time to upgrade the machines. If you really follow IG, you can see ImageGlass started with .NET 2.0, then 4.0, 4.6, 4.8. All the changes were announced at least 1 year. From 2.0 to 4.0, it was 3 years! Using .NET 4.8 (released in 2019) while Microsoft already released v6.0 makes IG outdated (you can read more about Microsoft .NET if you don't understand what I'm mentioning here).

If the requests like this keep going on, the technology will not grow fast.

**Comment by kutysam** [comment]
Sorry, but this has nothing to do with open source or closed source or .NET technology.

It's just about having an opt-in option on installation to let users control their preference and I have given my reasoning above. Starting up with computer is infact quite a important selectable option for a user to have.

> If the requests like this keep going on, the technology will not grow fast.
**Ultimately, your reasoning shows that for technology to grow fast all apps should adapt on opt out process for starting with windows, regardless on how often they are used. =/**

You haven't replied on **why you are not keen on putting it as an option in the checkbox during step 3**. [This is the feature that I am requesting] Compile the entire code to set such option is definetly a nogo for a retail user, don't you agree?
- It's already weird to force a user to start the app without their knowledge when they upgrade the app.
- If your reasoning is as per the first portion where, there is already an opt-out option for users, I won't go further in this discussion.

**Comment by d2phap** [comment]
> Ultimately, your reasoning shows that for technology to grow fast all apps should adapt on opt out process for starting with windows, regardless on how often they are used. =/

Yeah if you know what WinForms is, you should search the issues with "limitation" labeled. I don't think IG users use the app once a month, if so, they would stay with the default one.

> You haven't replied on why you are not keen on putting it as an option in the checkbox during step 3. [This is the feature that I am requesting] 

![image](https://user-images.githubusercontent.com/3154213/150461784-cedd6bc7-0b7e-4b41-b0c8-65078b94a897.png)


> Compile the entire code to set such option is definetly a nogo for a retail user, don't you agree?

Retail user will stay with the default app instead.


> - It's already weird to force a user to start the app without their knowledge when they upgrade the app.
> - If your reasoning is as per the first portion where, there is already an opt-out option for users, I won't go further in this discussion.

First, the initial purpose of developing ImageGlass is for myself, I'm the main user of the app.
Second, I have never been force people using ImageGlass, and the app is not to fit all people, too (and it's impossible). If they like it, they use it, or else, they will find another one. In fact, even though MS Photos is not so good, but I also use it when I need while having ImageGlass installed.

ImageGlass provides the flexible ways to change the default options to try to satisfy most of them. But there always is a small group of people who are not satisfied with the free app, I will just let them go, or they can modify the source code to build their own version. That's the point of open-source software.

**Comment by kutysam** [comment]
Once again, I'm looking at the perspective of all users and the feature I am requesting is a small feature which is kind to all users. There's no disadvantage to it, only advantages.

It's up to you to close this issue but there may be others whom wish to also comment on this. I'll leave it up to you since you are the owner.

I guess we agree to disagree.
My last 2 cents: You should enable all your apps on your computer to start with windows as you advocate it ;)

**Comment by Otiel** [comment]
I agree with @kutysam. Even if this new option is a good idea for some users who often start ImageGlass, it shouldn't be the default as it impacts the startup time of Windows. In my experience, few people read the release notes, so it should be an opt-in setting.

**Comment by ntvinh11586** [comment]
Agree with @kutysam, app should not be launched on windows startup in default. Nowadays, many apps have been defaultly launched in startup time, it is really annoying users to check then turn off most of them.

My suggestion to @d2phap: If ImageGlass actually need to run in background at startup time, I think it is better to optimise the app startup point to make it loaded faster.

**Comment by stedaniels** [comment]
I imagine the auto running in the background feature was added to increase the veracity (and profitability) of the Spider.com malware egress VPN feature that was also slipped in.
